File: sdks/csharp/src/NovaBilling/Types/BillableMetricResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record BillableMetricResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("id")]
    public required string Id { get; set; }

    [JsonPropertyName("name")]
    public required string Name { get; set; }

    [JsonPropertyName("code")]
    public required string Code { get; set; }

    [JsonPropertyName("description")]
    public string? Description { get; set; }

    [JsonPropertyName("aggregationType")]
    public required BillableMetricResponseAggregationType AggregationType { get; set; }

    [JsonPropertyName("fieldName")]
    public string? FieldName { get; set; }

    [JsonPropertyName("recurring")]
    public required bool Recurring { get; set; }

    [JsonPropertyName("filters")]
    public IEnumerable<BillableMetricFilterResponse> Filters { get; set; } =
        new List<BillableMetricFilterResponse>();

    [JsonPropertyName("createdAt")]
    public required string CreatedAt { get; set; }

    [JsonPropertyName("updatedAt")]
    public required string UpdatedAt { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/types/billable_metric_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class BillableMetricResponse < Internal::Types::Model
      field :id, -> { String }, optional: false, nullable: false
      field :name, -> { String }, optional: false, nullable: false
      field :code, -> { String }, optional: false, nullable: false
      field :description, -> { String }, optional: true, nullable: false
      field :aggregation_type, -> { Novabilling::Types::BillableMetricResponseAggregationType }, optional: false, nullable: false, api_name: "aggregationType"
      field :field_name, -> { String }, optional: true, nullable: false, api_name: "fieldName"
      field :recurring, -> { Internal::Types::Boolean }, optional: false, nullable: false
      field :filters, -> { Internal::Types::Array[Novabilling::Types::BillableMetricFilterResponse] }, optional: false, nullable: false
      field :created_at, -> { String }, optional: false, nullable: false, api_name: "createdAt"
      field :updated_at, -> { String }, optional: false, nullable: false, api_name: "updatedAt"
    end
  end
end
